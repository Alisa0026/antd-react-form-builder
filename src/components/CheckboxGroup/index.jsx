import React from 'react'
import _ from 'lodash'
import moment from 'moment'
import BasicItem from '../BasicItem'
import AntdCheckbox from 'antd/lib/checkbox'
import 'antd/lib/checkbox/style/css'
import FormItemComponentDecorator from '../../decorator/FormItemComponent'

const CheckboxGroup = AntdCheckbox.Group;

function component(BasicItemComponent){
  @FormItemComponentDecorator
  class FCheckboxGroup extends React.Component {

    render(){
      let { ...other } = this.props;
      other.targetComponent = CheckboxGroup;
      other.type = "checkboxgroup";
      this.propsAdapter(other);
      return (
        <BasicItemComponent { ...other }/>
      )
    }
    
  }
  return FCheckboxGroup;
}

export default component(BasicItem) 




