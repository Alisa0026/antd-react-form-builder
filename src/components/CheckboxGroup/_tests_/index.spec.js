import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import LocaleProvider from "antd/lib/locale-provider"
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import CheckboxGroup from '../index'

var options = [
  {
    value: "shenzhen",
    label: "深圳"
  },
  {
    value: "beijing",
    label: "北京"
  },
  {
    value: "hangzhou",
    label: "hangzhou"
  },
];

describe('CheckboxGroup',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <CheckboxGroup 
        required
        className="test"
        label="name"
        name="test"
        options={ options }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(CheckboxGroup); 
  it("props.options should equal options variable above",function(){
    const wrapper = mount(
      <CheckboxGroup 
        required
        name="test"
        label="label"
        options={ options }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(antdInputMount.props().options).toEqual(options);
  })
})

