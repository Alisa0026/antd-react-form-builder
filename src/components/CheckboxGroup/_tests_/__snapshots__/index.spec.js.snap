// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <div
          class="ant-checkbox-group test"
        >
          <label
            class="ant-checkbox-group-item ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              深圳
            </span>
          </label>
          <label
            class="ant-checkbox-group-item ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              北京
            </span>
          </label>
          <label
            class="ant-checkbox-group-item ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              hangzhou
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</span>
`;
